### ISSUE TITLE
The homepage is not displayed when no addon is added to primary hero shelf

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-server#15810]
STR:
When there's no primary hero addon added to the shelf from[ admin](https://addons-internal.stage.mozaws.net/en-US/admin/models/discovery/promotedaddon/?) the homepage shows a 500 error

![Server Error](https://user-images.githubusercontent.com/33448286/96856983-26cba780-1467-11eb-9662-ed1c3f6a157f.png)

This is reproducible on AMO stage - FF82(Win10).

**Comment by bobsilverberg** [comment]
This seems like more of a frontend issue, but maybe an API is returning a 500? Can you share anything from the console that might shed light on what the actual error is? What happens when another page is loaded first and then the home page is navigated to?

**Comment by bobsilverberg** [comment]
I did some digging and this is definitely a frontend error, but I did notice something odd with the API response as well. When there are no add-ons enabled for the primary hero shelf, the response still includes the fields for `addon`, but they are all empty. It looks like this:

```json
{
	"primary": {
		"addon": {
			"authors": [],
			"average_daily_users": null,
			"current_version": {
				"files": []
			},
			"guid": "",
			"name": null,
			"previews": [],
			"slug": "",
			"type": null,
			"promoted": {
				"category": null
			}
		},
		"external": {
			"guid": "",
			"homepage": null,
			"name": null,
			"type": null
		},
		"featured_image": ""
	},
	...
}
```

@eviljeff What do you think about that? Should we be returning that, or should `primary` just be `null`?

**Comment by eviljeff** [comment]
I can see why it returns an empty serialized response from looking at the code.  As to whether it should, I can't say - a null response would have to be handled too.

**Comment by bobsilverberg** [comment]
Yes, we'd have to handle either, but it just seems a bit odd to be receiving all this invalid data for an add-on. I suppose it's not a big deal, as this should never actually happen. I'll have frontend deal with this data as it currently stands.
